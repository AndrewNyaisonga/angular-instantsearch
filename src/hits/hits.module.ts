import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";

import { NgAisFooterModule } from "../footer/footer.module";
import { NgAisHeaderModule } from "../header/header.module";
import { NgAisHits } from "./hits";

@NgModule({
  declarations: [NgAisHits],
  entryComponents: [NgAisHits],
  exports: [NgAisHits],
  imports: [CommonModule, NgAisHeaderModule, NgAisFooterModule]
})
export class NgAisHitsModule {}
