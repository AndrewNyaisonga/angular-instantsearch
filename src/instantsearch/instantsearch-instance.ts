import { Injectable } from "@angular/core";
import instantsearch from "instantsearch.js/es";

@Injectable()
export class NgISInstance {
  public appId: string;
  public apiKey: string;
  public indexName: string;

  public numberLocale?: string;
  public searchFunction?: () => void;
  public createAlgoliaClient?: () => object;
  public searchParameters?: object | void;
  public urlSync?:
    | boolean
    | {
        mapping?: object;
        threshold?: number;
        trackedParameters?: string[];
        useHash?: boolean;
        getHistoryState?: () => object;
      };

  private instance?: {
    start: () => void;
  };

  public init(config: object) {
    this.instance = instantsearch(config);
  }

  public start() {
    this.instance.start();
  }
}
