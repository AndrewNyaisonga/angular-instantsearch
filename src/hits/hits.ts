import { Input, Component, ContentChild, TemplateRef } from "@angular/core";
import { connectHits } from "instantsearch.js/es/connectors";
import { isFunction } from "lodash";

import { BaseWidget } from "../base-widget";
import { NgAisInstance } from "../instantsearch/instantsearch-instance";
import { bem } from "../utils";

const cx = bem("Hits");

@Component({
  selector: "ng-ais-hits",
  template: `
    <div class="${cx()}">
      <ng-ais-header [header]="header" className="${cx(
        "header"
      )}"></ng-ais-header>

      <div class="${cx("body")}">
        <ng-container *ngTemplateOutlet="template; context: state"></ng-container>

        <!-- default rendering if no template specified -->
        <div *ngIf="!template">
          <ul class="${cx("list")}">
            <li
              class="${cx("item")}"
              *ngFor="let hit of hits"
            >
              <ng-ais-highlight attributeName="name" [hit]="hit">
              </ng-ais-highlight>
            </li>
          </ul>
        </div>
      </div>

      <ng-ais-footer [footer]="footer" className=${cx(
        "footer"
      )}></ng-ais-footer>
    </div>
  `
})
export class NgAisHits extends BaseWidget {
  @ContentChild(TemplateRef) public template;

  // render options
  @Input() transformItems?: Function;

  // inner widget state returned from connector
  public state = { hits: [] };

  get hits() {
    return isFunction(this.transformItems)
      ? this.transformItems(this.state.hits)
      : this.state.hits;
  }

  constructor(searchInstance: NgAisInstance) {
    super(searchInstance);
    this.createWidget(connectHits);
  }
}
