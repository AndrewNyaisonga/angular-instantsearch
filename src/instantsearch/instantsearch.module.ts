import { CommonModule } from "@angular/common";
import { ModuleWithProviders, NgModule } from "@angular/core";

import { NgISInstantSearch } from "./instantsearch";
import { NgISInstance } from "./instantsearch-instance";

@NgModule({
  declarations: [NgISInstantSearch],
  entryComponents: [NgISInstantSearch],
  exports: [NgISInstantSearch],
  imports: [CommonModule]
})
export class NgISInstantSearchModule {
  public static forRoot(): ModuleWithProviders {
    return {
      ngModule: NgISInstantSearchModule,
      providers: [NgISInstance]
    };
  }
}
