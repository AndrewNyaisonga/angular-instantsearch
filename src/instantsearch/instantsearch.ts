import { AfterViewInit, Component, Input, OnInit } from "@angular/core";

import { NgISInstance } from "./instantsearch-instance";

@Component({
  selector: "ngis-instantsearch",
  template: `<ng-content></ng-content>`
})
export class NgISInstantSearch implements AfterViewInit, OnInit {
  @Input() public config: InstantSearchConfig;

  constructor(private searchInstance: NgISInstance) {}

  public ngOnInit() {
    this.searchInstance.init(this.config);
  }

  public ngAfterViewInit() {
    this.searchInstance.start();
  }
}
