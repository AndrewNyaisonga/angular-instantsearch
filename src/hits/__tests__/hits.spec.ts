import { TestBed } from "@angular/core/testing";

import { NgAisInstantSearchModule } from "../../instantsearch/instantsearch.module";
import { NgAisHitsModule } from "../hits.module";
import { NgAisHits } from "../hits";

jest.mock("../../base-widget");

const defaultState = {
  hits: [
    { name: "foo", description: "foo" },
    { name: "bar", description: "bar" },
    { name: "foobar", description: "foobar" },
    { name: "barfoo", description: "barfoo" }
  ]
};

const render = (state?: {}) => {
  const fixture = TestBed.createComponent(NgAisHits);

  if (state) {
    fixture.componentInstance.updateState({ ...defaultState, ...state }, false);
  }

  fixture.detectChanges();
  return fixture;
};

describe("Hits", () => {
  beforeEach(() =>
    TestBed.configureTestingModule({
      declarations: [],
      imports: [NgAisInstantSearchModule.forRoot(), NgAisHitsModule]
    })
  );

  it("renders markup without state", () => {
    const fixture = render();
    expect(fixture).toMatchSnapshot();
  });

  it("renders markup with state", () => {
    const fixture = render({});
    expect(fixture).toMatchSnapshot();
  });
});
