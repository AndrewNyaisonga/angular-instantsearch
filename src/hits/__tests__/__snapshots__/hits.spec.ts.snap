// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits renders markup with state 1`] = `
<ng-ais-hits
  cx={[Function Function]}
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  template="undefined"
  updateState={[Function Function]}
>
  
      <div
    class="ais-Hits"
  >
    
        
    <ng-ais-header
      ng-reflect-class-name="ais-Hits-header ais-header"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-Hits-body ais-body"
    >
      
          
      
      
  
          
      
          
      
      <div>
        
            
        <ul
          class="ais-Hits-list"
        >
          
              
          
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                foo
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                bar
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                foobar
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                barfoo
              </span>
            </ng-ais-highlight>
            
              
          </li>
          
            
        </ul>
        
          
      </div>
      
        
    </div>
    
  
        
    <ng-ais-footer
      ng-reflect-class-name="ais-Hits-footer ais-footer"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-hits>
`;

exports[`Hits renders markup without state 1`] = `
<ng-ais-hits
  cx={[Function Function]}
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  template="undefined"
  updateState={[Function Function]}
>
  
      <div
    class="ais-Hits"
  >
    
        
    <ng-ais-header
      ng-reflect-class-name="ais-Hits-header ais-header"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-Hits-body ais-body"
    >
      
          
      
      
  
          
      
          
      
      <div>
        
            
        <ul
          class="ais-Hits-list"
        >
          
              
          
          
            
        </ul>
        
          
      </div>
      
        
    </div>
    
  
        
    <ng-ais-footer
      ng-reflect-class-name="ais-Hits-footer ais-footer"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-hits>
`;

exports[`Hits should apply \`transformItems\` if specified 1`] = `
<ng-ais-hits
  cx={[Function Function]}
  searchInstance={[Function NgAisInstance]}
  state={[Function Object]}
  template="undefined"
  transformItems={[Function Function]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-Hits"
  >
    
        
    <ng-ais-header
      ng-reflect-class-name="ais-Hits-header ais-header"
    >
      
      
      
      
    
    </ng-ais-header>
    
  
        
    <div
      class="ais-Hits-body ais-body"
    >
      
          
      
      
  
          
      
          
      
      <div>
        
            
        <ul
          class="ais-Hits-list"
        >
          
              
          
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - foo
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - bar
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - foobar
              </span>
            </ng-ais-highlight>
            
              
          </li>
          <li
            class="ais-Hits-item"
          >
            
                
            <ng-ais-highlight
              attributename="name"
              ng-reflect-attribute-name="name"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - barfoo
              </span>
            </ng-ais-highlight>
            
              
          </li>
          
            
        </ul>
        
          
      </div>
      
        
    </div>
    
  
        
    <ng-ais-footer
      ng-reflect-class-name="ais-Hits-footer ais-footer"
    >
      
      
      
      
    
    </ng-ais-footer>
    
      
  </div>
    
</ng-ais-hits>
`;
